modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/model/impl/FriendlyURLEntryImpl.java

```diff
@@ -107,4 +110,26 @@ public boolean isMain() throws PortalException {
 		return false;
 	}
 
+	private long _getCompanyId(AssetEntry assetEntry) {
+		long companyId = CompanyThreadLocal.getCompanyId();
+
+		if (companyId != 0) {
+			return companyId;
+		}
+
+		if (assetEntry != null) {
+			return assetEntry.getCompanyId();
+		}
+
+		FriendlyURLEntry friendlyURLEntry =
+			FriendlyURLEntryLocalServiceUtil.fetchFriendlyURLEntry(
+				getFriendlyURLEntryId());
+
+		if (friendlyURLEntry != null) {
+			return friendlyURLEntry.getCompanyId();
+		}
+
+		return companyId;
+	}
```

Comment:
The friendly url entry already has a `getCompanyId()` method, so all this logic should not be necessary. Are you doing this for any reason?

Reply:
yeah, 

we do not know on this case if the process is from. So on most of the cases, the the company can come from the CompanyThreadLocal.getCompanyId(). This should be the company from we are working on.

In the cases where the value is 0, SaaS or upgrades, we have to obtain a posible candidate.

the assetEntry is obtained and used on the main method (if pertinent) so the cost to use it is not relevant.

But the friendlyURLEntry was never used, so this is a new call to the database. this should not be a performance problem, but I prefer to delay the use to be the last option.

That was my reason to do this this way

If you think otherwise, what is your proposal?

Reply:
I don't understand. Do you mean that the company ID of the friendly url entry may be uninitialized? In what cases will the friendly url entry company ID be different from the thread local company ID (assuming thread local company ID != 0).

My proposal is to simply do:
```
private long _getCompanyId(AssetEntry assetEntry) {
    return this.getCompanyId();
}
```
and inline.

Reply:
no, I was thinking that the CompanyThreadLocal.getCompanyId() could not be initilizated no the friendlyURLEntry

But, I just realized that I do not need to search it!! I just realized that I'm inside the object!!! Thanks for pointing at it!

Reply:
Just a tought... can we access to a company from another company...??? there is any case of it?

serious question 🤔

Reply:
thanks for the info :)